lib/plugins/aws/common/index.js

```diff
@@ -46,15 +45,13 @@ class AwsCommon {
     };
 
     this.hooks = {
-      'aws:common:validate:validate': () => BbPromise.bind(this).then(this.validate),
+      'aws:common:validate:validate': async () => this.validate(),
 
-      'aws:common:cleanupTempDir:cleanup': () => BbPromise.bind(this).then(this.cleanupTempDir),
+      'aws:common:cleanupTempDir:cleanup': async () => this.cleanupTempDir(),
 
-      'aws:common:moveArtifactsToPackage:move': () =>
-        BbPromise.bind(this).then(this.moveArtifactsToPackage),
+      'aws:common:moveArtifactsToPackage:move': async () => this.moveArtifactsToPackage(),
 
-      'aws:common:moveArtifactsToTemp:move': () =>
-        BbPromise.bind(this).then(this.moveArtifactsToTemp),
+      'aws:common:moveArtifactsToTemp:move': async () => this.moveArtifactsToTemp(),
```

Comment:
Hooks can be _sync_. I'd leave them not marked as _async_ functions. Let's leave this (whether it's _async_ or _sync_) up to functions that are invoked